plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreReadAccessOperations.java

```diff
@@ -0,0 +1,340 @@
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT));
+    }
+
+    @Test
+    public void testSelectViewsWithFilterByTableName()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.views WHERE table_name = 'test_table_0'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT));
+    }
+
+    @Test
+    public void testSelectViewsWithLikeOverTableName()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.views WHERE table_name LIKE 'test%'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT));
+    }
+
+    @Test
+    public void testSelectColumnsWithoutPredicate()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_TABLE, TEST_ALL_TABLES_COUNT));
+    }
+
+    @Test
+    public void testSelectColumnsFilterByInformationSchema()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_schema = 'information_schema'", ImmutableMap.of());
+    }
+
+    @Test
+    public void testSelectColumnsFilterBySchema()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_schema = 'test_schema_0'",
+                ImmutableMap.of(
+                        GET_ALL_TABLES_WITH_PARAMETER, 1,
+                        GET_ALL_VIEWS_WITH_PARAMETER, 1,
+                        GET_TABLE, TEST_TABLES_IN_SCHEMA_COUNT));
+    }
+
+    @Test
+    public void testSelectColumnsWithLikeOverSchema()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_schema LIKE 'test%'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_TABLE, TEST_ALL_TABLES_COUNT));
+    }
+
+    @Flaky(issue = "For particular query there are 10294+/-1 GET_TABLE requests, therefore it is flaky", match = ".*")
+    @Test
+    public void testSelectColumnsFilterByTableName()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_name = 'test_table_0'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
```

Comment:
TEST_SCHEMAS_COUNT since the calls are `O(#schema)`
(using TEST_SCHEMAS_COUNT != TEST_TABLES_IN_SCHEMA_COUNT would help)

Reply:
Yeah, my mistake, changing `TEST_TABLES_IN_SCHEMA_COUNT` to different value helps to see a problem